devel/check-k8s.sh

```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+# shellcheck disable=SC1091
+source ./devel/include/verbose.inc
+
+if [ "${OC_USERNAME}" != "" ] && [ "${OC_PASSWORD}" != "" ] && [ "${OC_API_URL}" != "" ]; then
+    oc login -u "${OC_USERNAME}" -p "${OC_PASSWORD}" "${OC_API_URL}" \
+    || die "Failed to log in the cluster"
+else
+    oc whoami &>/dev/null \
```

Comment:
You are right mate. But reviewing the change I can see something that is confusing and arise some concerns:
- `pre-commit.sh` and the pipeline are calling to `./devel/lint.sh`. Internally, `lint.sh` is calling to `./devel/check-k8s.sh` when it is possible too.
- `pre-commit.sh` force the execution of `./devel/check-k8s.sh`, so it pushes the developer to be logged in the cluster to validate the objects; this could be a problem if in some moment there are not connectivity with a cluster (such as making changes off-line or with no Internet connections when we are using a no local cluster).
- When the conditions to validate the k8s objects are met, the validation will be launched twice (once forced by pre-commit.sh a second time from lint.sh script), so it is duplicated. I don't like this duplicity.

I think is more appropriate to validate the objects only when the checks can be made, and display a warning message from the `lint.sh` script when the check can not be made. I am going to accommodate that change and update the documentation related with this in DEVOPS.md file.

With the above considerations, the final behaviour would be:
- Only `lint.sh` launches the k8s object validation which is called from the pipeline and from the `pre-commit.sh` hook.
- When `lint.sh` can not launch `./devel/check-k8s.sh` because not logged in in a cluster or not credentials are set up (that is the scenario in the pipeline), a warning message is printed out, but it doesn't break the execution of the `lint.sh` script, neither the pipeline.
- The developer can use `./devel/check-k8s.sh` whenever it is needed, and it will fail if no logged in a cluster or no credentials are provided.

Thanks @frasertweedale , your comment made me meditate.

Reply:
Thanks @avisiedo, I think it's a good change.  Merging now.